unskript-ctl/unskript_ctl_factory.py

```diff
@@ -134,17 +134,26 @@ def update_credentials_to_uglobal(self):
                 creds_json_files.extend(glob.glob(pattern, recursive=True))
                 break
         self.creds_json_files = creds_json_files
+        c_data = {}
         for creds_json_file in creds_json_files:
             with open(creds_json_file, 'r', encoding='utf-8') as f:
-                c_data = json.load(f)
-                if c_data.get('metadata').get('connectorData') == '{}':
-                    continue 
-                mapping[c_data.get('metadata').get('type')] = {"name": c_data.get('metadata').get('name'), 
+                try:
+                    c_data = json.load(f)
+                except:
```

Comment:
This doesnt sound correct, if the json file is corrupted, how do we catch that? Empty should be a specific exception check

Reply:
Implemented empty file check and exception handling in case json decoding fails for a given credential file.